src/System.IO.FileSystem/tests/Directory/Modify_FailSafe.cs

```diff
@@ -0,0 +1,581 @@
+        if (!"bar".Equals(dirs[0].Name))
+            throw new Exception("DirectoryInfo's Name should have been bar, but was: " + dirs[0].Name);
+
+        FileSystemInfo[] infos = di.GetFileSystemInfos();
+        if (infos.Length != 2)
+            throw new Exception("GetFileSystemInfos should have returned 2!  got: " + infos.Length);
+        FileInfo tempFi = infos[0] as FileInfo;
+        DirectoryInfo tempDi = null;
+        if (tempFi == null)
+        {
+            tempFi = infos[1] as FileInfo;
+            tempDi = infos[0] as DirectoryInfo;
+        }
+        else
+        {
+            tempDi = infos[1] as DirectoryInfo;
+        }
+        if (!tempFi.Name.Equals("foo"))
+            throw new Exception("GetFileSystemInfo returned FileInfo with wrong name!  got: " + tempFi.Name);
+        if (!tempDi.Name.Equals("bar"))
+            throw new Exception("GetFileSystemInfo returned DirectoryInfo with wrong name!  got: " + tempDi.Name);
+
+
+        // Test DirectoryInfo.Name on something like "c:\bar\"
+        DirectoryInfo subDir = new DirectoryInfo(Path.Combine(di.Name, "bar\\"));
+        if (!subDir.Name.Equals("bar"))
+            throw new Exception("Subdirectory name was wrong.  Expected bar, Got: " + subDir.Name);
+
+        DirectoryInfo parent = subDir.Parent;
+        if (!DirNameEquals(parent.FullName, di.FullName))
+            throw new Exception("DI.FullName != SubDir.Parent.FullName!  subdir full name: " + parent.FullName);
+
+        // Check more info about the DirectoryInfo
+        String rootName = Path.GetPathRoot(Directory.GetCurrentDirectory());
+        DirectoryInfo root = di.Root;
+        if (!rootName.Equals(root.Name))
+            throw new Exception(String.Format("Root directory name was wrong!  rootName: {0}  DI.Root.Name: {1}", rootName, root.Name));
+
+        // Test DirectoryInfo behavior for "c:\"
+        DirectoryInfo c = new DirectoryInfo("c:\\");
+        if (!"c:\\".Equals(c.Name))
+            throw new Exception("DirectoryInfo name for c:\\ was wrong!  got: " + c.Name);
+        if (!"c:\\".Equals(c.FullName))
+            throw new Exception("DirectoryInfo FullName for c:\\ was wrong!  got: " + c.FullName);
+        if (null != c.Parent)
+            throw new Exception("DirectoryInfo::Parent for c:\\ is not null!");
+
+        FailSafeDirectoryOperations.DeleteDirectoryInfo(di, true);
+        di.Refresh();
+        if (di.Exists)
+            throw new Exception("Directory still exists at end of test!");
+
+        Assert.True(s_pass);
+    }
+
+    private static bool DirNameEquals(String a, String b)
+    {
+        if (a.Length > 3 && a[a.Length - 1] == Path.DirectorySeparatorChar)
+            a = a.Substring(0, a.Length - 1);
+        if (b.Length > 3 && b[b.Length - 1] == Path.DirectorySeparatorChar)
+            b = b.Substring(0, b.Length - 1);
+        return a.Equals(b);
+    }
+
+    [Fact]
+    [ActiveIssue(0)] // SetCurrentDirectory
```

Comment:
Need something here other than just ActiveIssue(0).  Either file an issue or delete the test.